### DISCUSSION TITLE
Continue workflow during fanout

### DISCUSSION THREAD

**Discussion by jrowinski3d** [argoproj/argo-workflows#7722] [Q&A] [answered]
Hi, I'm wondering if it is possible to continue a workflow regardless if the fanout has **finished**

Something along the lines of where C1 can start when B1 has finished. But in this scenario B2/B3 still are processing.
D1 would have to wait for all of C1/2/3 to finish to complete the fan in process

```bash
#           *
#           | 
#          A1 
#           | 
#        /  |  \ 
#       B1  B2  B3 
#       |   |    |
#       C1
#       *   *   *
#        \  |  /
#         \ | /
#           D1
```

**Comment by terrytangyuan** [comment, marked as answer]
You can use `depends` to specify the dependencies between those tasks. https://github.com/argoproj/argo-workflows/blob/master/docs/enhanced-depends-logic.md

**Reply by jrowinski3d** [reply]
Hey @terrytangyuan , wouldn't that still be deterministic by `depends: B` which would wait for the full fanout to finish? Unless I'm misunderstanding, but B1/B2/B3 can be dynamic and have many more task sets within it depending on the result of `A`.

**Reply by terrytangyuan** [reply]
Could you group them together in a way and then specify the dependencies among the groups? Similar to https://github.com/argoproj/argo-workflows/blob/master/examples/dag-diamond-steps.yaml

**Reply by jrowinski3d** [reply]
Is there a possibility to handle grouping of `fan-out` strategy? I guess this would mean that the first initial task could output a list of lists to handle that?

But if you want it to handle groups, those have to be pre-defined correct?

**Reply by terrytangyuan** [reply]
Yes those groups can be pre-defined templates.

**Reply by jrowinski3d** [reply]
Sorry but want to confirm with you that there is no way to handle this **dynamically** during a fan-out approach?

**Reply by terrytangyuan** [reply]
I think https://github.com/argoproj/argo-workflows/issues/3171 might be what you are looking for but it's not supported yet. Currently using enhanced depends you'll have to specify the specific task names explicitly.

**Reply by jrowinski3d** [reply]
ahhh yes, exactly what I am looking for. Thanks for your time @terrytangyuan, really appreciate it. Will keep my eye open on that one.

**Comment by jrowinski3d** [comment]
Marking as **answered** based on this proposal: https://github.com/argoproj/argo-workflows/issues/3171

**Reply by hello-willy** [reply]
hi jrowinski3d:
    I also have this problem. Have you solved this problem？
    I use the following settings：
```yaml
depends: "bp.AnySucceeded"
```
   but it does not work！How did you solve this problem? could you share the answer? thank you!!!

**Reply by jrowinski3d** [reply]
hey @hello-willy , we also had a discussion in a stackoverflow that solved it. See the following link: https://stackoverflow.com/questions/70960203/argo-workflow-to-continue-processing-during-fan-out

@crenshaw-dev was kind enough to answer it

**Reply by hello-willy** [reply]
nice!! thank you!
i will try it right way!!

**Reply by hello-willy** [reply]
hi @jrowinski3d 
  IS WORK!! thank you VERY MUCH.
